src/main/java/com/nutomic/syncthingandroid/fragments/DeviceSettingsFragment.java

```diff
@@ -110,7 +115,9 @@ public void onCreate(Bundle savedInstanceState) {
     @Override
     public void onSaveInstanceState(Bundle outState) {
         super.onSaveInstanceState(outState);
-        outState.putSerializable("device", mDevice);
+        if (mIsCreate && mDevice != null) {
+            outState.putSerializable("device", mDevice);
```

Comment:
I think `Device` needs to implement `Serializable` for this. 

Another option would be functions `Device.save(Bundle bundle)` and `Device.restore(Bundle bundle)` (because I don't know how well serializable works).